stack-graphs/src/stitching.rs

```diff
@@ -482,6 +483,8 @@ impl PathStitcher {
             queue: VecDeque::new(),
             next_iteration,
             cycle_detector: CycleDetector::new(),
+            // By default, there's no artificial bound on the amount of work done per phase
+            max_work_per_phase: usize::MAX,
```

Comment:
> So maybe the parameter should be optional, or there is a bounded and unbounded version of the methods?

I made it so that the default is the previous unbounded behavior, and that we only bound the work if you explicitly call the setter function